dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -306,6 +306,9 @@ public abstract class Entry<S extends Entry, T extends Version> implements Compa
     @Transient
     private Set<OrcidAuthor> orcidAuthors = new HashSet<>();
 
+    @Column(nullable = false)
+    private boolean wasEverPublic;
```

Comment:
The rationale is that "published" has a particular meaning in our system.  In the future, other things may happen that make a entry "public" (maybe it is shared in a special way), upon which we can set the field, and thus it is a different concept and has a different name.

Reply:
> The rationale is that "published" has a particular meaning in our system. In the future, other things may happen that make a entry "public" (maybe it is shared in a special way), upon which we can set the field, and thus it is a [slightly] different concept and has a different name.

That said, a different word might be better.  We're trying to prevent the deletion of Entries that have been used/referenced by the public, in the interest of FAIRness, and in this implementation, we're using potential exposure as a proxy.